src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dba/CassandraConfigurationManager.scala

```diff
@@ -0,0 +1,88 @@
+package com.microsoft.partnercatalyst.fortis.spark.dba
+import java.util.concurrent.ConcurrentHashMap
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{BlacklistedTerm, Geofence, SiteSettings}
+import com.microsoft.partnercatalyst.fortis.spark.sources.streamprovider.ConnectorConfig
+import org.apache.spark.SparkContext
+import com.datastax.spark.connector._
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import scala.compat.java8.FunctionConverters._
+
+@SerialVersionUID(100L)
+class CassandraConfigurationManager extends ConfigurationManager with Serializable with Loggable {
+  // Note: trusted sources are cached for the lifetime of the configuration manager since in order to update them,
+  // streaming must be restarted (and hence the configuration manager would be replaced).
+  private lazy val connectorToTrustedSources = new ConcurrentHashMap[String, Seq[String]]()
+
+  override def fetchConnectorConfigs(sparkContext: SparkContext, pipeline: String): List[ConnectorConfig] = {
+    def fetchTrustedSources(connectorName: String): Seq[String] = {
+      sparkContext.cassandraTable(CassandraSchema.KeyspaceName, CassandraSchema.Table.TrustedSourcesName)
+        .select("externalsourceid")
+        .map(row => row.getString("externalsourceid")).collect()
+    }
+
+    val pipelineConfigRows = sparkContext.cassandraTable[CassandraSchema.Table.Stream](CassandraSchema.KeyspaceName,
+      CassandraSchema.Table.StreamsName).collect()
```

Comment:
No biggie, but I think we could get rid of (or defer) this `collect()` call since we call `toList` at the end - this way we only iterate through the list once.